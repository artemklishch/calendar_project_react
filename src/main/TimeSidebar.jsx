import React from 'react';

const TimeSidebar = () => {
  const arrOfTimePoints = Array(25)
    .fill(null)
    .map((elem, index) => {
      if(index !== 0 && index !== 24){
        elem = `${index}:00`;
      } 
      return elem;
    });
  return (
    arrOfTimePoints.map(timePoint =>
      <div key={Math.random().toFixed(5)} className="main__sidebar_timing_place">
        <span className="main__sidebar_time">{timePoint}</span>
        <div className="main__sidebar_timing_place-bord"></div>
      </div>)
  );
};
export default TimeSidebar;