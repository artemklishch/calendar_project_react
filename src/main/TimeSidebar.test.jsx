import React from 'react';
import { shallow } from 'enzyme';
import TimeSidebar from './TimeSidebar';

import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
Enzyme.configure({ adapter: new Adapter() });
import toJson from 'enzyme-to-json';

describe('TimeSidebar', () => {
  it('should display time sidebar', () => {
    const wrappedComponent = shallow(<TimeSidebar />);
    expect(toJson(wrappedComponent)).toMatchSnapshot();
  })
});