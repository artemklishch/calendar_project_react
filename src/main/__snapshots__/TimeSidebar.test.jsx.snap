// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimeSidebar should display time sidebar 1`] = `
Array [
  <div
    className="main__sidebar_timing_place"
    key="0.45044"
  >
    <span
      className="main__sidebar_time"
    />
    <div
      className="main__sidebar_timing_place-bord"
    />
  </div>,
  <div
    className="main__sidebar_timing_place"
    key="0.54503"
  >
    <span
      className="main__sidebar_time"
    >
      1:00
    </span>
    <div
      className="main__sidebar_timing_place-bord"
    />
  </div>,
  <div
    className="main__sidebar_timing_place"
    key="0.79675"
  >
    <span
      className="main__sidebar_time"
    >
      2:00
    </span>
    <div
      className="main__sidebar_timing_place-bord"
    />
  </div>,
  <div
    className="main__sidebar_timing_place"
    key="0.07249"
  >
    <span
      className="main__sidebar_time"
    >
      3:00
    </span>
    <div
      className="main__sidebar_timing_place-bord"
    />
  </div>,
  <div
    className="main__sidebar_timing_place"
    key="0.68958"
  >
    <span
      className="main__sidebar_time"
    >
      4:00
    </span>
    <div
      className="main__sidebar_timing_place-bord"
    />
  </div>,
  <div
    className="main__sidebar_timing_place"
    key="0.22775"
  >
    <span
      className="main__sidebar_time"
    >
      5:00
    </span>
    <div
      className="main__sidebar_timing_place-bord"
    />
  </div>,
  <div
    className="main__sidebar_timing_place"
    key="0.74606"
  >
    <span
      className="main__sidebar_time"
    >
      6:00
    </span>
    <div
      className="main__sidebar_timing_place-bord"
    />
  </div>,
  <div
    className="main__sidebar_timing_place"
    key="0.45675"
  >
    <span
      className="main__sidebar_time"
    >
      7:00
    </span>
    <div
      className="main__sidebar_timing_place-bord"
    />
  </div>,
  <div
    className="main__sidebar_timing_place"
    key="0.97224"
  >
    <span
      className="main__sidebar_time"
    >
      8:00
    </span>
    <div
      className="main__sidebar_timing_place-bord"
    />
  </div>,
  <div
    className="main__sidebar_timing_place"
    key="0.20340"
  >
    <span
      className="main__sidebar_time"
    >
      9:00
    </span>
    <div
      className="main__sidebar_timing_place-bord"
    />
  </div>,
  <div
    className="main__sidebar_timing_place"
    key="0.72207"
  >
    <span
      className="main__sidebar_time"
    >
      10:00
    </span>
    <div
      className="main__sidebar_timing_place-bord"
    />
  </div>,
  <div
    className="main__sidebar_timing_place"
    key="0.81216"
  >
    <span
      className="main__sidebar_time"
    >
      11:00
    </span>
    <div
      className="main__sidebar_timing_place-bord"
    />
  </div>,
  <div
    className="main__sidebar_timing_place"
    key="0.97964"
  >
    <span
      className="main__sidebar_time"
    >
      12:00
    </span>
    <div
      className="main__sidebar_timing_place-bord"
    />
  </div>,
  <div
    className="main__sidebar_timing_place"
    key="0.27349"
  >
    <span
      className="main__sidebar_time"
    >
      13:00
    </span>
    <div
      className="main__sidebar_timing_place-bord"
    />
  </div>,
  <div
    className="main__sidebar_timing_place"
    key="0.58127"
  >
    <span
      className="main__sidebar_time"
    >
      14:00
    </span>
    <div
      className="main__sidebar_timing_place-bord"
    />
  </div>,
  <div
    className="main__sidebar_timing_place"
    key="0.12490"
  >
    <span
      className="main__sidebar_time"
    >
      15:00
    </span>
    <div
      className="main__sidebar_timing_place-bord"
    />
  </div>,
  <div
    className="main__sidebar_timing_place"
    key="0.64781"
  >
    <span
      className="main__sidebar_time"
    >
      16:00
    </span>
    <div
      className="main__sidebar_timing_place-bord"
    />
  </div>,
  <div
    className="main__sidebar_timing_place"
    key="0.56286"
  >
    <span
      className="main__sidebar_time"
    >
      17:00
    </span>
    <div
      className="main__sidebar_timing_place-bord"
    />
  </div>,
  <div
    className="main__sidebar_timing_place"
    key="0.18732"
  >
    <span
      className="main__sidebar_time"
    >
      18:00
    </span>
    <div
      className="main__sidebar_timing_place-bord"
    />
  </div>,
  <div
    className="main__sidebar_timing_place"
    key="0.31400"
  >
    <span
      className="main__sidebar_time"
    >
      19:00
    </span>
    <div
      className="main__sidebar_timing_place-bord"
    />
  </div>,
  <div
    className="main__sidebar_timing_place"
    key="0.04831"
  >
    <span
      className="main__sidebar_time"
    >
      20:00
    </span>
    <div
      className="main__sidebar_timing_place-bord"
    />
  </div>,
  <div
    className="main__sidebar_timing_place"
    key="0.48606"
  >
    <span
      className="main__sidebar_time"
    >
      21:00
    </span>
    <div
      className="main__sidebar_timing_place-bord"
    />
  </div>,
  <div
    className="main__sidebar_timing_place"
    key="0.23720"
  >
    <span
      className="main__sidebar_time"
    >
      22:00
    </span>
    <div
      className="main__sidebar_timing_place-bord"
    />
  </div>,
  <div
    className="main__sidebar_timing_place"
    key="0.23119"
  >
    <span
      className="main__sidebar_time"
    >
      23:00
    </span>
    <div
      className="main__sidebar_timing_place-bord"
    />
  </div>,
  <div
    className="main__sidebar_timing_place"
    key="0.23479"
  >
    <span
      className="main__sidebar_time"
    />
    <div
      className="main__sidebar_timing_place-bord"
    />
  </div>,
]
`;
